docs/tutorials/shallow-join.rst

```diff
@@ -0,0 +1,359 @@
+computation across a distributed set of resources. Spark manages the
+lineage of a block of transformed data so that if a node goes down,
+Spark can restart the computation for just the missing blocks.
+
+`Jupyter Notebook <https://github.com/jupyter/notebook>`__ is an interactive web interface
+for a kernel. Jupyter allows you quickly prototype by writing code in runnable cells, computing a final result
+as you go. Visualization of data is also easily done through integration Jupyter cell 'magics' to create JavaScript and
+HTML outputs.
+
+Here, we will combine these two services to demonstrate an operation we are naming "Shallow Join". This operation
+is a means of imposing a small, covering set of data, onto a much larger set of data. In our example, GDELT data has
+geometries for each event, but it is not directly known in which country the event took place. We "join" this geometry
+against the polygons of the covering set in order to calculate statistics for a geographical region.
+
+
+Prerequisites
+-------------
+
+.. warning::
+
+    You will need access to a Hadoop |hadoop_version| installation with Yarn as well as an Accumulo |accumulo_version| database.
+
+    You will need to have ingested GDELT data using GeoMesa. Instructions are available in :doc:`geomesa-examples-gdelt`.
+
+    You will need to have ingested a shapefile of polygons outlining your regions of choice. In this tutorial we use
+    `this <http://thematicmapping.org/downloads/world_borders.php>` shapefile of countries.
+
+You will also need:
+
+-  a `Spark <http://spark.apache.org/>`__ 1.5.0 or later distribution
+-  an Accumulo user that has appropriate permissions to query your data
+-  `Java JDK 8 <http://www.oracle.com/technetwork/java/javase/downloads/index.html>`__,
+-  a `Jupyter Notebook <https://github.com/jupyter/notebook>`__ server with the
+    `Apache Toree https://toree.incubator.apache.org/documentation/user/installation.html`__ Scala kernel installed
+
+
+Create RDDs
+^^^^^^^^^^^
+
+First, set up the parameters and initialize each of the desired data stores.
+
+.. code-block:: scala
+
+    val gdeltParams = Map(
+      "instanceId" -> "instance",
+      "zookeepers" -> "zoo1,zoo2,zoo3",
+      "user"       -> "user",
+      "password"   -> "*****",
+      "auths"      -> "USER,ADMIN",
+      "tableName"  -> "geomesa_catalog")
+
+    val countriesParams = Map(
+      "instanceId" -> "instance",
+      "zookeepers" -> "zoo1,zoo2,zoo3",
+      "user"       -> "user",
+      "password"   -> "*****",
+      "auths"      -> "USER,ADMIN",
+      "tableName"  -> "countries_catalog")
+
+    val gdeltDs = DataStoreFinder.getDataStore(gdeltParams)
+    val countriesDs = DataStoreFinder.getDataStore(countriesParams)
+
+
+Next, initialize a Spark context from the Simple Feature Types of the data stores.
+
+.. code-block:: scala
+    val types = gdeltDs.getTypeNames.map(ds.getSchema) ++ countriesDs.getTypeNames.map(ds.getSchema)
```

Comment:
Needs to be an empty line between the code-block directive and the first line of the block.